department-of-veterans-affairs/vets-website
# Income and asset 86209 associated incomes

## THIS FORM IS DISABLED IN PRODUCTION

# Income and Asset Statement Form (21P-0969)

## Description
Add a new section with the [Array Builder Pattern(Multiple responses list & loop)](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/patterns/array-builder/README.md)

## Issue
[Section IV: Recurring income and net worth associated with financial accounts #86209](https://github.com/department-of-veterans-affairs/va.gov-team/issues/86209)

### Related Issues
n/a

### Epic
[Digitize Income and Asset Statement Form 21P-0969](https://app.zenhub.com/workspaces/benefits-pension-64e775aaa6b1ca1ed49b2ede/issues/gh/department-of-veterans-affairs/va.gov-team/80925)

### Screenshots
| Intro Page | Income Relationship Page | Income Type Page | Summary Page |
| ------- |  ------- |  ------- |  ------- |
|![localhost_3001_income-and-asset-statement-form-21p-0969_ (4)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/d7074139-1e43-4437-8017-30c3bc07efa1)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (5)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/a40e18dd-e0cc-4135-847a-fe40b4afdb19)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (7)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/d457cf43-5b42-40a1-bdaf-54af5468d2f9)|![localhost_3001_income-and-asset-statement-form-21p-0969_ (8)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/7107fcd4-8457-48b9-88df-505a39a68582)

## How to run in local environment
1. Check out this branch locally in 'vets-website'
2.  Checkout this [branch](https://github.com/department-of-veterans-affairs/content-build/pull/2075) locally in 'content-build'
3. Run 'vets-api' in your local environment
4. Run the 'vets-website' app with `yarn watch --env entry=static-pages,auth,login-page,profile,dashboard,income-and-asset-statement`

## How to toggle feature flag in local environment
1. Go to [income_and_assets_form_enabled](http://localhost:3000/flipper/features/income_and_assets_form_enabled)

## How to verify
1. Go to `http://localhost:3001/income-and-asset-statement-form-21p-0969` in your browser
2. Sign in with a mock user and 'Start application'
3. Go to the 'Associated incomes' chapter
4. Verify that required fields are required
5. Verify that optional fields are optional
6. Verify that all fields follow validation rules
7. Verify that up to 5 associated incomes can be added, edited and deleted
8. Verify that the associated incomes are visible and editable on the 'Review and submit' page
9. Verify that the form can be saved via the api endpoint
10. Verify that the form can be submitted

## Testing

## What areas of the site does it impact?
Income and Asset Statement Application

### Quality Assurance & Testing

### Error Handling

### Authentication
